src/Simulation/Simulators.Tests/SerializationTests/StateSerializationTests.cs

```diff
@@ -1,70 +1,62 @@
-            {
-                NQubits = 2,
-                Table = new StabilizerState.TableArray
+            NQubits = 2,
+            Table = new StabilizerState.TableArray(
+                Data: new List<bool>
                 {
-                    Data = new List<bool>
-                    {
-                        true, false, false, false, true, false
-                    },
-                    Dimensions = new List<int> { 2, 3 }
-                }
-            };
-            var json = JsonSerializer.Serialize<Microsoft.Quantum.Experimental.State>(state);
-            var expectedJson = @"{
-                ""n_qubits"": 2,
-                ""data"": {
-                    ""Stabilizer"": {
-                        ""n_qubits"": 2,
-                        ""table"": {
-                            ""v"": 1,
-                            ""dim"": [2, 3],
-                            ""data"": [true,false,false,false,true,false]
-                        }
+                    true, false, false, false, true, false
+                },
+                Dimensions: new List<int> { 2, 3 }
+            )
+        };
+        var json = JsonSerializer.Serialize<Microsoft.Quantum.Experimental.State>(state);
+        var expectedJson = @"{
+            ""n_qubits"": 2,
+            ""data"": {
+                ""Stabilizer"": {
+                    ""n_qubits"": 2,
+                    ""table"": {
+                        ""v"": 1,
+                        ""dim"": [2, 3],
+                        ""data"": [true,false,false,false,true,false]
                     }
                 }
-            }";
+            }
+        }";
 
-            expectedJson.AssertJsonIsEqualTo(json);
-        }
+        expectedJson.AssertJsonIsEqualTo(json);
+    }
 
-        [Fact]
-        public void StabilizerArrayDeserializesCorrectly()
-        {
-            var expectedJson = @"
-                {
-                    ""v"": 1,
-                    ""dim"": [2, 3],
-                    ""data"": [true,false,false,false,true,false]
-                }
-            ";
-            var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(expectedJson));
-            var array = JsonSerializer.Deserialize<StabilizerState.TableArray>(ref reader);
-            var state = new StabilizerState
+    [Fact]
+    public void StabilizerArrayDeserializesCorrectly()
+    {
+        var expectedJson = @"
             {
-                NQubits = 2,
-                Table = array
-            };
-        }
+                ""v"": 1,
+                ""dim"": [2, 3],
+                ""data"": [true,false,false,false,true,false]
+            }
+        ";
+        var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(expectedJson));
+        var array = JsonSerializer.Deserialize<StabilizerState.TableArray>(ref reader);
+        var state = new StabilizerState
+        {
+            NQubits = 2,
+            Table = array
+        };
     }
```

Comment:
What do the lines 56 - 60 do? They create an instance that is never used? In order to make sure that no exception thrown during `new StabilizerState`?

Reply:
Let me look into this, I suspect this is a mistake made in the original experimental version of the feature, such that I can likely correct it as part of this PR. Thanks for the catch!